### DISCUSSION TITLE
OMNI mode for note on/off for live use with controller with faders.

### DISCUSSION THREAD

**Discussion by bearwolf** [probonopd/MiniDexed#288] [Ideas]
Allright my usecase might be a bit slim but hear me out. 
I play live and thought it would be so cool to use the minidexed as a main synth for some smaller gigs and a backup for others since it's so conveniently sized. 

So my main midi controller is a keylab essential61 that i control mainstage with usually. It has conveniently 8 faders and 8 encoders (plus one of each for a "master channel") 

So what i would like to do is map the eight engines out on different channels and mix between them with the faders. Like electric piano on ch1, and then in the chorus add some strings by pushing up the volume on ch2 with fader 2 on my keyboard. Only problem is my keyboard (and most others from what i know) don't send note on/off or most data on omni channels, just 1-16. 

So i'm stuck having multiple keyboards or changing the midi channel a lot. I would like to just have the 8 engines always playing with the volumes individually controlled. 

i'm guessing it would involve modifying the mididevice.cpp [case MIDI_NOTE_ON:] to have a flag that ignores the channel or sends to multiple. But i'm a hack and not about to make my own builds just yet :)

Optimally i mean if every engine hade a channel setting individually for both note on/off and control change that would be some next level customization. 

But if there would be a fork build that just does it for live usage or a one off flag in the performance.ini that turns off channel filtering for the note on/off it would be so helpful! 

Pic of me adjusting ch2 volume:
![IMG_3820](https://user-images.githubusercontent.com/3290087/174658316-0dbe43d1-e52b-44c3-85ea-29e1186459f2.jpg)

Again, rather slim usecase, most people use the minidexed with sequencers and recording i guess so menudiving isn't a big problem for some things. But wouldn't it be cool to roll up to a gig, plug this little synth in and go to town a full gig without having to change any patches with 8 sounds all ready to play and mix at your fingertips :)

Thanks for reading my ramblings!

**Comment by craigyjp** [comment]
Why not just use pgm changes, why make it so complicated. I think what you are trying to do is impossible as the faders will all transmit on the same channel.

**Reply by bearwolf** [reply]
I don't want to make 40 songs of pgm change setlists for every gig as a backup when i can just just use this way instead. 

The faders are already adjusting the corresponding volume for the eight channels, and i have the encoders above for cutoff for every channel. They can all be indivdually set to a midi channel and cc adress. So everything is already working as intended except i'd need to route the midi through some external stuff to get note on/off on the channels i'd like. 

This is a pretty common way of playing keyboards live, especially for worship people from what i've seen online so i'd say it's the opposite of complicated :D play everything and volume up/down per instrument. ezpz ;)

**Reply by craigyjp** [reply]
Ok but, 8 faders, 8 sounds, so group the sounds in the same way 1-8 and you can quickly go up and down through the sounds with pgm changes or am I missing something. 

I just don't see away around it with faders all on the same channel which is normal for most controllers. 

You could build a translation box that reads in each controller and transmits volume changes in channels 1-8, a simple Arduino box can do that with simple translation of cc message numbers to cc 07 on different channels.

**Reply by bearwolf** [reply]
I think we are misunderstanding eachoter :) The volume changes already work perfectly. I can control all 8 volumes perfectly fine already.

I made a short video better explaining what i mean: https://youtu.be/AyNZq5I236g 

All i want is midichannel per engine not affecting the incoming notes, or a separate function so i can have midi ch1 playing notes on engine 1-4 and midi ch2 going to engine 5-8 even though they all get their cc separately. 

Heres an image i borrowed of a popular worship sound kit for mainstage (i don't use it but i have similar setups for some bands) 
as you can see everything is layered but there is much focus on live mixing between the engines dynamically. 
![image](https://user-images.githubusercontent.com/3290087/174665652-2779181f-7643-4446-8e01-217e0ecff82f.png)

**Reply by craigyjp** [reply]
Ok, a little more complicated, I still think you could do it with an Arduino or Teensy to read in your notes on/off and send on all 8 channels.

As you have already set the minidexed to 8 channels in order to use the volume controls, then I can't see any other way around it. 

I built a midi merge that is capable of doing this on only 5 channels, taking a single channel in and creating upto 5 splits on different outgoing channels, of course these splits could over lap and be the whole keyboard range. If you removed all the other merger stuff and extra functions I'm sure you could expand this to 8 outputs with some code changes.

**Reply by bearwolf** [reply]
Yeah exactly, there are some smart midi filters too out there that could do this. But adding much more equipment to it kind of defeats the "oh i have a backup synth in my back pocket phew" factor of just plugging in the keyboard power and a linebox :D  that's why i wanted to ask here in the discussion about the idea and not make an issue. 

Thanks for bearing with me!

**Comment by probonopd** [comment]
Wouldn't https://github.com/probonopd/MiniDexed/issues/63 allow for this?

>  I would like to just have the 8 engines always playing with the volumes individually controlled.

You would assign the volume of each TG to one slider each.
Simple (once #63 would be implemented), or am I missing something?

**Reply by craigyjp** [reply]
If it can do this, read 8 controllers on one channel and assign them to the volume of the 8 TGs then yes, bit it's only 8 sounds, you would need to switch performances or pgm to get more. The OP mentioned 40 sounds per set.

**Reply by bearwolf** [reply]
I think it could! If i could start with all engines on midichannel 1, then learn them to other midi channels via the learn feature and it didn't mess with the original midi channel for the engine then yes! Or if learn is implemented i could also use some empty CC commands on midichannel one to control all engines on a separate CC. I added a link above to a short video where i maybe explained better exactly what i mean :  https://youtu.be/AyNZq5I236g

I don't necesarily need 40 sounds, it's more likely 6 sounds in different combinations, a piano, some strings, a decent pad, a lead synth, a bass synth to add some oomf. 
In my main setlist in mainstage i have all songs set up with the exact splits and stuff per song and so on. Lots of work. But as a backup? Beatles eh? piano sound up. Ballad? add the strings channel to the piano. Funk? clav channel upp, maybe add strings in the chorus for some phat chords. It's a great way to work fast on stage :)

**Reply by bearwolf** [reply]
Any news on #63 btw @probonopd ? i Saw that the person working on it had to tap out of development at some stage because life/work. But that midi .ini file would be so sweet and allow so much good customization :)

**Reply by probonopd** [reply]
Agree that would be a great addition to have; eagely waiting for a C++ skilled volunteer to take up the challenge. Unfortunately I don't have more information than what is stated in the ticket.
